alacritty/src/config/window.rs

```diff
@@ -74,11 +74,31 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        let (lines, columns) = (self.dimensions.lines, self.dimensions.columns.0);
+        let (lines_is_none_zero, columns_is_non_zero) = (lines != 0, columns != 0);
+
+        // return dimensions if both `lines` and `columns` is non-zero
+        if lines_is_none_zero && columns_is_non_zero {
             Some(self.dimensions)
+        }
+        // Warn if either `columns` or `lines` is non-zero
```

Comment:
```suggestion
        // Warn if either `columns` or `lines` is non-zero.
```

Just put this documentation inside the `else if` block. Like this you're completely messing up formatting making the code itself incredibly difficult to read.

The `if` branch documentation can also go inside if you want consistency.